qln/msghandler.go

```diff
@@ -53,6 +47,28 @@ func (nd *LitNode) PeerHandler(msg lnutil.LitMsg, q *Qchan, peer *RemotePeer) er
 
 }
 
+// BroadcastMessage sends a chat message to all connected websocket clients
+func (nd *LitNode) BroadcastChatMessage(msg lnutil.LitMsg) error {
+	chat, ok := msg.(lnutil.ChatMsg)
+	if !ok {
+		return fmt.Errorf("can't cast to chat message")
+	}
+
+	// get the current set of channels and save them, they may be deleted by litrpc
+	// before the iteration is over
+	var channels []chan lnutil.ChatMsg
```

Comment:
wouldn't two loops make the problem worse? Instead of ranging over the map of UserWsCons and sending a message to each, this ranges over UserWsCons, caches each one, then starts over and ranges over them to send messages.  It seems like there's more time for things to close during the 'seek back to the start' part.